src/SourceBuild/content/test/Microsoft.DotNet.Tests/PackageSourceMappingsTests.cs

```diff
@@ -0,0 +1,242 @@
+                BuildWithOnlineFeeds = useOnlineFeeds,
+                SourceBuildSources = sources,
+                CustomSources = customSources
+            };
+
+            if (sourceBuild)
+            {
+                task.ReferencePackagesSourceName = "reference-packages";
+                task.PreviouslySourceBuiltSourceName = "previously-source-built";
+                task.PrebuiltSourceName = "prebuilt";
+            }
+
+            task.Execute();
+
+            TokenizeLocalNugetConfigFeeds(modifiedNugetConfig);
+
+            string expectedNugetConfigContents = File.ReadAllText(expectedNugetConfig);
+            string modifiedNugetConfigContents = File.ReadAllText(modifiedNugetConfig);
+            Assert.Equal(expectedNugetConfigContents, modifiedNugetConfigContents);
+        }
+
+        private static void UpdateNugetConfigTokens(string nugetConfigFile)
+        {
+            string fileContents = File.ReadAllText(nugetConfigFile);
+            foreach (KeyValuePair<string, string> kvp in TestSetup.LocalTokenSourceMappings)
+            {
+                fileContents = fileContents.Replace(kvp.Key, kvp.Value);
+            }
+            File.WriteAllText(nugetConfigFile, fileContents);
+        }
+
+        private static void TokenizeLocalNugetConfigFeeds(string nugetConfigFile)
+        {
+            string fileContents = File.ReadAllText(nugetConfigFile);
+            foreach (KeyValuePair<string, string> kvp in TestSetup.LocalTokenSourceMappings)
+            {
+                fileContents = fileContents.Replace(kvp.Value, kvp.Key);
+            }
+            File.WriteAllText(nugetConfigFile, fileContents);
+        }
+
+        internal class PackageSourceMappingsSetup
+        {
+            private static PackageSourceMappingsSetup? instance;
+            private static readonly object myLock = new();
+            private Dictionary<string, string>? localTokenSourceMappings;
+
+            public static readonly string PackageSourceMappingsRoot = Path.Combine(Directory.GetCurrentDirectory(), nameof(PackageSourceMappingsTests));
+
+            private readonly string ArcadeSource = Path.Combine(PackageSourceMappingsRoot, "arcade");
+            private readonly string RuntimeSource = Path.Combine(PackageSourceMappingsRoot, "runtime");
+            private readonly string PreviouslySourceBuiltSource = Path.Combine(PackageSourceMappingsRoot, "previously-source-built");
+            private readonly string ReferencePackagesSource = Path.Combine(PackageSourceMappingsRoot, "reference-packages");
+            private readonly string PrebuiltSource = Path.Combine(PackageSourceMappingsRoot, "prebuilt");
+            private readonly string SourceBuildReferencePackagesSource = Path.Combine(PackageSourceMappingsRoot, "source-build-reference-package-cache");
+
+            public readonly string SourceBuildReferencePackagesRepoDir = Path.Combine(PackageSourceMappingsRoot, "sbrpDir");
```

Comment:
Why the `Dir` suffix?

Reply:
No reason, really. It represents test SBRP repo directory. I'll remove the suffix.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/44649/commits/f13f4f1cfb5a7adec023571eaf49ddcc8a051648